### ISSUE TITLE
Feature Request: Add support for OAuth

### ISSUE THREAD

**Issue by fullstackpotato** [ciur/papermerge#210]
Very much similar to the reasoning behind #94 

As LDAP authentication is getting added, would it be possible to also add in support for OAuth? The reasoning is pretty much identical to the ticket mentioned above, having support for OAuth would allow users to hook up to pre-existing Single Sign On's (e.g. Keycloak).

**Comment by ciur** [comment]
@0x01BM, thank you for opening this feature request. It makes perfect sense. I didn't know about keycloak project - thanks for mentioning it. I will definitely add OAuth integration.

**Comment by Auridh** [comment]
@ciur 

I saw OAuth2 and OpenID [mentioned](https://github.com/ciur/papermerge/issues/548) as a feature for 3.0.
So are you able to hook up pre-existing SSOs like Keycloak or Authentik now?
I tried looking it up in the documentation, but the [section for OAuth2](https://docs.papermerge.io/3.0/setup/docker-compose/#oauth-20) is still empty.

**Comment by ciur** [comment]
OAuth2 support is "almost there". By "almost there" I mean that I added the feature, I tested it once with [Github](https://github.com/papermerge/auth-server/blob/main/auth_server/auth.py#L52) and [Google](https://github.com/papermerge/auth-server/blob/main/auth_server/auth.py#L44) as auth servers, and it worked!

However, since then I've never used it again and not yet documented it.
Also, I have not yet tried it with Keycloak/Authentik/Authelia etc.
The plan is to have support for OAuth2, OpenID (and LDAP).

In short: documentation coming soon.

**Comment by ciur** [comment]
Feature landed (OIDC support) in 3.2: https://docs.papermerge.io/3.2/sso/oidc/overview/
